client/src/components/Ticketing/ticketingmanager/Event/components/OptionUpdateTable.tsx

```diff
@@ -56,13 +58,10 @@ export const OptionUpdateTable = (props: OptionTableProps) => {
             {options && (
               <FormButton
                 title='Add Row'
-                className='text-green-500 disabled:text-gray-300'
+                className='text-green-500 hover:text-green-600 disabled:text-gray-300'
                 testID='add-row-button'
-                onClick={() => {
-                  arrayHelpers.insert(0, getOption(options[0]));
-                  setOptions(options.slice(1, options.length));
-                }}
-                disabled={options.length === 0}
+                onClick={() => addRow(options, setOptions)}
+                disabled={disabled(options)}
```

Comment:
The only issue there is that if the event restriction is on it will allow for the possibility of adding more rows than there are events available and leaving a row with no options to select from.

Reply:
Ok, well in that case, do you think it'd be possible to have the criteria be on the minimum of `number of rows` and then also `number of events` if the Event Restriction is on?